### PR TITLE
Scope

### PR DESCRIPTION
**Pull request by adejam**
The objective of this article is to explain the concept of Scope in Javascript.

The PR does the following:

- Explain a Block of code
- Explain Global Scope
- Explain Block Scope
- Explain Scope pollution in Javascript.
- Explain the best practice of Scoping in Javascript.

### CODE REVIEW

File: scope.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+A block of code could be a function, like below:
+
+```js
+const tellThePeriod = () => {
+  let period = "morning";
+  console.log(period); // morning
+};
+```
+
+We can Notice that the function body is a block of code.
```
Comment (bolah2009) [inline scope.md]: can we use `notice` not `Notice`

File: scope.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+- We first declared a variable `number` with the `let` keyword which means the variable can be reassigned to another value.
+
+- Inside the function `logNumber()`, we mistakenly reassigned the value of `number` to `157` perhaps we intended to name the variable as something else.
+
+- When `logNumber()` was called, `number` gets reassigned to `157`, which affected the global variable `num`
+
+- We won’t get an error as reassignment is allowed here, so if we decided to use `number` later on, we’ll be using the new value of `number`.
+
+With this, it is better not to always make every variable to be globally scoped.
```
Comment (bolah2009) [inline scope.md]: There are exceptions where this is might be your intended/expected result. Or maybe provide a better approach to the issue. 

Also, can you explain further examples of scopes? e.g. you can talk about what happens when a variable of the same name is both declared outside and inside of a block, would this be an error? If no, which takes precedence?

File: scope.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+Block scoped variables allow defining variables with precision, and not pollute the global namespace hence If a variable does not need to exist outside a block— it shouldn’t!.
+
+## For extra reading:
+
+[W3schools](https://www.w3schools.com/js/js_scope.asp)
```
Comment (bolah2009) [inline scope.md]: Would be better to also take a look at other articles to get a wider understanding of this topic.

**Review by bolah2009 — APPROVED** [review]
Looks good to me. 🚀 

Please attend to the comments below before merging.

File: scope.md (hunk 2 above)
Comment (adejam) [reply scope.md]: I'll add an example which explains this

**Comment by adejam** [comment]
Thanks for the review. I'll make changes where necessary

**Pull request merged**